File: clojure/cljfind/test/cljfind/filetypes_test.clj
(ns cljfind.filetypes-test
  (:require [clojure.test :refer :all])
  (:use [cljfind.filetypes :only (get-file-type)]))

(deftest test-get-file-type-archive-file
  (let [f "archive.zip"]
    (testing "test-get-file-type-archive-file"
      (is (= (get-file-type f) :archive)))))

(deftest test-get-file-type-audio-file
  (let [f "music.mp3"]
    (testing "test-get-file-type-audio-file"
      (is (= (get-file-type f) :audio)))))

(deftest test-get-file-type-binary-file
  (let [f "binary.exe"]
    (testing "test-get-file-type-binary-file"
      (is (= (get-file-type f) :binary)))))

(deftest test-get-file-type-code-file
  (let [f "code.clj"]
    (testing "test-get-file-type-code-file"
      (is (= (get-file-type f) :code)))))

(deftest test-get-file-type-font-file
  (let [f "font.ttf"]
    (testing "test-get-file-type-font-file"
      (is (= (get-file-type f) :font)))))

(deftest test-get-file-type-image-file
  (let [f "image.png"]
    (testing "test-get-file-type-image-file"
             (is (= (get-file-type f) :image)))))

(deftest test-get-file-type-text-file
  (let [f "text.txt"]
    (testing "test-get-file-type-text-file"
             (is (= (get-file-type f) :text)))))

(deftest test-get-file-type-video-file
  (let [f "movie.mp4"]
    (testing "test-get-file-type-video-file"
             (is (= (get-file-type f) :video)))))

(deftest test-get-file-type-xml-file
  (let [f "markup.xml"]
    (testing "test-get-file-type-xml-file"
      (is (= (get-file-type f) :xml)))))

(deftest test-get-file-type-unknown-file
  (let [f "unknown.xyz"]
    (testing "test-get-file-type-unknown-file"
      (is (= (get-file-type f) :unknown)))))

File: javascript/jsfind/tests/filetypes.test.js
/*
 * filetypes_test.js
 *
 * Some tests of filetypes.js
 */

const {FileType} = require('../src/filetype');
const {FileTypes} = require('../src/filetypes');

describe('testing filetypes', () => {
    it('testFileTypesArchiveFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'archive.zip';
        fileTypes.isArchiveFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.ARCHIVE);
        });
    });

    it('testFileTypesAudioFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'music.mp3';
        fileTypes.isAudioFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.AUDIO);
        });
    });

    it('testFileTypesBinaryFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'binary.exe';
        fileTypes.isBinaryFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.BINARY);
        });
    });

    it('testFileTypesCodeFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'code.js';
        fileTypes.isCodeFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.CODE);
        });
    });

    it('testFileTypesFontFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'font.ttf';
        fileTypes.isFontFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.FONT);
        });
    });

    it('testFileTypesImageFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'image.png';
        fileTypes.isImageFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.IMAGE);
        });
    });

    it('testFileTypesTextFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'text.txt';
        fileTypes.isTextFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.TEXT);
        });
    });

    it('testFileTypesVideoFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'movie.mp4';
        fileTypes.isVideoFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.VIDEO);
        });
    });

    it('testFileTypesXmlFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'markup.xml';
        fileTypes.isXmlFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.XML);
        });
    });

    it('testFileTypesUnknownFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'unknown.xyz';
        fileTypes.isUnknownFile(filename).then((res) => {
            expect(res).toBeTruthy();
        });
        fileTypes.getFileType(filename).then((type) => {
            expect(type).toBe(FileType.UNKNOWN);
        });
    });
});
